crates/swc_ecma_minifier/src/compress/pure/array.rs

```diff
@@ -0,0 +1,28 @@
+use swc_atoms::js_word;
+use swc_common::util::take::Take;
+use swc_ecma_ast::*;
+use swc_ecma_utils::ExprFactory;
+
+use super::Pure;
+
+impl Pure<'_> {
+    /// new Array(...) -> Array(...)
+    pub(super) fn remove_array_new(&mut self, e: &mut Expr) {
+        match e {
+            Expr::New(NewExpr {
+                span,
+                callee,
+                args,
+                type_args,
+            }) if callee.as_ident().map(|x| &x.sym) == Some(&js_word!("Array")) => {
```

Comment:
Possibly here need to check ident syntax context equality to unresolved_syntax_context, to prevent remove new form this case

```javascript
class Array {}

new Array(); // here Array is not a global Array object
```

Reply:
Can't figure out why I'm getting the wrong result.
```JavaScript
new Array();
```

```JavaScript
class Array {}
new Array();
```

In both cases I get Mark(0) as their syntax context.
But the unresolved is Mark(1).

@kdy1 
Is there something wrong with the test case, or with my code?

```rust
span.ctxt.outer() == self.marks.unresolved_mark
span.ctxt == self.expr_ctx.unresolved_ctxt
```
I have tried both of the above methods. No luck so far.